addons/web/static/src/js/views/list/list_editable_renderer.js

```diff
@@ -404,71 +408,82 @@ ListRenderer.include({
      * @returns {Promise}
      */
     setRowMode: function (recordID, mode) {
-        var self = this;
-        var record = self._getRecord(recordID);
-        if (!record) {
-            return Promise.resolve();
+        // Use a mutex because we don't want to rerender a row before the previous render is finished
+        let mutex = this.rowModeMutexesByRecordId.get(recordID);
+        if (!mutex) {
+            mutex = new concurrency.Mutex();
+            this.rowModeMutexesByRecordId.set(recordID, mutex);
         }
 
-        var editMode = (mode === 'edit');
-        var $row = this._getRow(recordID);
-        this.currentRow = editMode ? $row.prop('rowIndex') - 1 : null;
-        var $tds = $row.children('.o_data_cell');
-        var oldWidgets = _.clone(this.allFieldWidgets[record.id]);
-
-        // Prepare options for cell rendering (this depends on the mode)
-        var options = {
-            renderInvisible: editMode,
-            renderWidgets: editMode,
-        };
-        options.mode = editMode ? 'edit' : 'readonly';
-
-        // Switch each cell to the new mode; note: the '_renderBodyCell'
-        // function might fill the 'this.defs' variables with multiple promise
-        // so we create the array and delete it after the rendering.
-        var defs = [];
-        this.defs = defs;
-        _.each(this.columns, function (node, colIndex) {
-            var $td = $tds.eq(colIndex);
-            var $newTd = self._renderBodyCell(record, node, colIndex, options);
-
-            // Widgets are unregistered of modifiers data when they are
-            // destroyed. This is not the case for simple buttons so we have to
-            // do it here.
-            if ($td.hasClass('o_list_button')) {
-                self._unregisterModifiersElement(node, recordID, $td.children());
+        let self = this;
+        let defs = [];
+
+        return mutex.exec(() => {
+            var record = self._getRecord(recordID);
```

Comment:
to avoid indentation, create a function a put it in as the mutex callback
see basicmodel.js: save function

Reply:
Done :+1: